### PR TITLE
fix #2066 - wrong trivial accessor correction with instance_eval

### LINKED ISSUE #2066: TrivialAccessors gives wrong correction with instance_eval
Consider this snippet of code:
```ruby
obj.instance_eval do
  def foo
    @foo
  end
end
```
This code shouldn't be autocorrected with `attr_reader` because the method definition is in instance context. `attr_reader` doesn't exist in this context.

### PR DESCRIPTION
**Pull request by tranvictor**

### CODE REVIEW

File: lib/rubocop/cop/style/trivial_accessors.rb [hunk 1]
```diff
@@ -117,13 +117,19 @@ def accessor(kind, method_name)
...
-          if node.type == :def
+          if node.type == :def && !in_instance_context?(node)
             autocorrect_instance(node)
           elsif node.type == :defs && node.children.first.type == :self
             autocorrect_class(node)
           end
         end
 
+        def in_instance_context?(node)
+          (node.parent &&
```
Comment (bbatsov) [inline lib/rubocop/cop/style/trivial_accessors.rb]: Do you see a `node.parent` repetition here?

File: lib/rubocop/cop/style/trivial_accessors.rb [hunk 2]
```diff
@@ -117,13 +117,19 @@ def accessor(kind, method_name)
...
+           node.parent.type == :block &&
+           node.parent.children.first.children.last == :instance_eval)
```
Comment (bbatsov) [inline lib/rubocop/cop/style/trivial_accessors.rb]: I'm not fond of using array access to node elements - we favour the use of splatting.

File: spec/rubocop/cop/style/trivial_accessors_spec.rb [hunk 3]
```diff
@@ -349,6 +349,24 @@
...
+        [
+          'obj.instance_eval do |_|',
+          '  def foo',
+          '    @foo',
+          '  end',
+          'end'
+        ]
+      end
+
+      it 'does not autocorrect' do
```
Comment (bbatsov) [inline spec/rubocop/cop/style/trivial_accessors_spec.rb]: You shouldn't care about auto-correct at all - simply check there are no offenses.

**Comment by bbatsov** [comment]
The commit message should follow the project's style and refer the issue report. You also need to add a changelog entry.

**Comment by tranvictor** [comment]
@bbatsov I'm sorry that I didn't follow the pull request convention, I will modify the message and add changelog entry.

File: spec/rubocop/cop/style/trivial_accessors_spec.rb (hunk 3 above)
Comment (tranvictor) [inline spec/rubocop/cop/style/trivial_accessors_spec.rb]: Do you mean that I shouldn't check if it's in instance_eval here, just check in the offense detection?

Comment (bbatsov) [inline spec/rubocop/cop/style/trivial_accessors_spec.rb]: I mean you should just check whether this code produces offences or not.

**Pull request closed without merging**